common/djangoapps/student/admin.py

```diff
@@ -149,20 +149,43 @@ class Meta(object):
     exclude = ('dashboard_tracking_code',)
 
 
+class CourseEnrollmentForm(forms.ModelForm):
+
+    def __init__(self, *args, **kwargs):
+        super(CourseEnrollmentForm, self).__init__(*args, **kwargs)
+
+        if self.data.get('course'):
+            self.data['course'] = CourseKey.from_string(self.data['course'])
```

Comment:
Is there a reason you don't need the try..except block that you use below when you user `CourseKey.from_string()` here?